src/app-layer-ftp.c

```diff
@@ -446,6 +795,32 @@ void RegisterFTPParsers(void)
 
         AppLayerParserRegisterGetStateProgressCompletionStatus(ALPROTO_FTP,
                                                                FTPGetAlstateProgressCompletionStatus);
+
+
+        AppLayerRegisterExpectationProto(IPPROTO_TCP, ALPROTO_FTPDATA);
+        AppLayerParserRegisterParser(IPPROTO_TCP, ALPROTO_FTPDATA, STREAM_TOSERVER,
+                                     FTPDataParseRequest);
+        AppLayerParserRegisterParser(IPPROTO_TCP, ALPROTO_FTPDATA, STREAM_TOCLIENT,
+                                     FTPDataParseResponse);
+        AppLayerParserRegisterStateFuncs(IPPROTO_TCP, ALPROTO_FTPDATA, FTPDataStateAlloc, FTPDataStateFree);
+        AppLayerParserRegisterParserAcceptableDataDirection(IPPROTO_TCP, ALPROTO_FTPDATA, STREAM_TOSERVER | STREAM_TOCLIENT);
+        AppLayerParserRegisterTxFreeFunc(IPPROTO_TCP, ALPROTO_FTPDATA, FTPDataStateTransactionFree);
+        AppLayerParserRegisterDetectStateFuncs(IPPROTO_TCP, ALPROTO_FTPDATA, FTPDataStateHasTxDetectState,
+                FTPDataGetTxDetectState, FTPDataSetTxDetectState);
+
+        AppLayerParserRegisterGetFilesFunc(IPPROTO_TCP, ALPROTO_FTPDATA, FTPDataStateGetFiles);
+
+        AppLayerParserRegisterGetTx(IPPROTO_TCP, ALPROTO_FTPDATA, FTPDataGetTx);
+
+        AppLayerParserRegisterGetTxCnt(IPPROTO_TCP, ALPROTO_FTPDATA, FTPDataGetTxCnt);
+
+        AppLayerParserRegisterGetStateProgressFunc(IPPROTO_TCP, ALPROTO_FTPDATA, FTPDataGetAlstateProgress);
+
+        AppLayerParserRegisterGetStateProgressCompletionStatus(ALPROTO_FTPDATA,
+                FTPDataGetAlstateProgressCompletionStatus);
+
+        sbcfg.buf_size = 4096;
```

Comment:
seems the file data itself is excluded from the memcap. To include it register alloc/free funcs here for the streaming api

Reply:
I got the idea but don't manage to understand what to do here. Can you tell me which functions to use ?

Reply:
see https://github.com/OISF/suricata/blob/master/src/app-layer-htp.c#L2274

The idea is to register your own alloc/free funcs that track memuse/memcap

Reply:
Do you want me to introduce a new memcap or should I use the reassembly one ?

Reply:
Not sure what reassembly has to do with it.

Whats needed is a FTP specific memcap.

Reply:
Ok :+1: